cmd/dep/ensure.go

```diff
@@ -145,17 +210,11 @@ func (cmd *ensureCommand) Run(ctx *dep.Ctx, args []string) error {
 			Constraint: pc.Constraint,
 		}
 
-		if p.Lock != nil {
-			for i, lp := range p.Lock.P {
-				if lp.Ident() == pc.Ident {
-					p.Lock.P = append(p.Lock.P[:i], p.Lock.P[i+1:]...)
-					break
-				}
-			}
-		}
+		// Ignore the lockfile for this dependency and allow its version to change
+		params.ToChange = append(params.ToChange, pc.Ident.ProjectRoot)
```

Comment:
I checked back through blame, and it looks like the original intent was indeed doing what you're doing here. It was work done back in early Dec, and we were flying pretty seat-of-pants at that point; I just hadn't explained the nuance of using `params.ToChange` instead of just pulling it out of the lock.